network/wsPeer.go

```diff
@@ -440,6 +476,15 @@ func (wp *wsPeer) readLoop() {
 		msg.processing = wp.processed
 		msg.Received = time.Now().UnixNano()
 		msg.Data = slurper.Bytes()
+		if msg.Tag == protocol.ProposalPayloadTag &&
+			wp.features&vfCompressedProposal != 0 &&
+			bytes.Equal(msg.Data[:4], zstdCompressionMagic[:]) {
+			msg.Data, err = zstd.Decompress(nil, msg.Data)
```

Comment:
(Prompted by Chris mentioning this PR:)

It might be worth using the `zstd.NewReader().Read()` API to make sure we don't decompress more than the max expected size of a payload.  An adversary might be able to supply a short compressed payload that decompresses to something huge (gigabytes), causing the receiver to run out of memory and exit.